### ISSUE TITLE
Templates need upgrade

### ISSUE THREAD

**Issue by sjimenez77** [devonfw/production-line#38]
I have realized that some of the templates are not properly upgraded. For example the node versions for **devon4ng** and **devon4node** are still **12.16.1**, far from the current LTS one **14.15.5**. 

Please, check this and review the other templates.

**Comment by maybeec** [comment]
I even would go one step further as of our current target of supporting ci/CD out of the box, I would like to focus on code first configuration approach including different proposed version scheme. Currently we are setting up jenkins configuration by these templates, which in my view more and more is an anti pattern.

**Comment by sjimenez77** [comment]
Indeed, I fully agree and I would like to focus on IaC configuration with terraform, ansible, etc. but, as long as we have to support Jenkins in the PL instances, we would still need to provide the necessary configuration inside the templates. 

At the same time I'd rather work on Kubernetes templates to deploy our stacks and even adding the necessary files to add a CICD namespace with the configuration and common toolchain to build, test, monitor, etc. 

Definitely a topic for the next DA.

**Comment by maybeec** [comment]
Let's please stay in close contact in regards to that. IaC is on our roadmap for 2021 and I would like to harvest without starting another ivory tower setup like shop floor which just for example worked if your are cluster owner.
I am happy that you are willing to contribute in this direction.
